### ISSUE TITLE
"Focus State Internal Error - A focus node has isFocusedLeaf: true, yet it has enabled children." when root node has a single child with its first child disabled

### ISSUE THREAD

**Issue by LBRDan** [jamesplease/lrud#78]
ATM when FocusRoot register the starting root node, it does so by setting `isFocusedLeaf: true`, as it's the first mounting node inside the tree

Following the order of the component mount (parent to children) and keeping in mind the useEffect execution order (children to parent - siblings in order of render) it seems like mounting some specific FocusNodes tree structure some there could be a specific time when the error in object occurs

CodeSandbox [Here](https://codesandbox.io/s/please-root-isfocusedleaf-repro-simple-qsrfn)

(Please, read after running the CodeSandbox above)
Inside `getChildren` function, the current inspecting node's children gets filtered via `nodeCanReceiveIndirectFocus` ([here](https://github.com/jamesplease/lrud/blob/f912d681d9b1a20c3b01dcb5b050151b43fe2609/src/utils/tree-navigation.ts#L58))

This function filters out recursively the node children mainly looking at `isDisabled` or `isExiting` (etc...)

As the first time it register a node (non root), it will result in having no children valid focusable children (when there is one in truth) and it returns the current focus hierarchy [here](https://github.com/jamesplease/lrud/blob/f912d681d9b1a20c3b01dcb5b050151b43fe2609/src/utils/tree-navigation.ts#L64)

In this case, inside the first node register computation, the focused leaf state should be transferred to root node's first and only child, then down to its correct (second) child

This could "maybe" be a corner case, or a false positive, but it indeed happens that in a specific brief moment, there is something odd about the leaf currently in focus

**Comment by jamesplease** [comment]
Thanks for the issue. Some notes (mostly jotting this down for myself)

- While the tree is being built up, the root refuses to pass focus down to the parent until B exists. This is because when the parent only has A, A is disabled so there's no leaf that can be focused.
- The state validator is naive...it only looks at the root's children and sees that the parent is there, and not disabled. The validator needs to recursively dig to make sure that the node can actually receive focus.

tl;dr I think it's a bug with the validation checker rather than a bug with the way that the lib is working.

**Comment by jamesplease** [comment]
Fixed in `0.0.18`. Let me know if this works for you @LBRDan !
